src/lib/Types/Core.hs

```diff
@@ -1981,10 +1984,17 @@ instance GenericE TopFunBinding where
 instance SinkableE TopFunBinding
 instance HoistableE  TopFunBinding
 instance SubstE Name TopFunBinding
-instance SubstE (AtomSubstVal CoreIR) TopFunBinding
 instance AlphaEqE TopFunBinding
 instance AlphaHashableE TopFunBinding
 
+instance SubstE (AtomSubstVal CoreIR) TopFunBinding where
```

Comment:
Did you figure out why we even need this instance? I would expect that TopFunBinding would need `SubstE Name` and nothing else.